File: trunk/Projects/Framework-Samples/PLINQO/CSharp/Tracker/Source/Tracker.Tests/QueryTests/OperatorTests.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using CodeSmith.Data.Linq;
using NUnit.Framework;
using Tracker.Core.Data;

namespace Tracker.Tests.QueryTests
{
    [TestFixture]
    public class OperatorTests : TestBase
    {       
        [Test]
        public void Contains()
        {
            var db = new TrackerDataContext { Log = Console.Out };

            var assignedId = new List<int?> { null, 1 };
            // null and int
            var q1 = db.Task.ByAssignedId(assignedId);

            var assignedList = q1.ToList();
            Assert.IsNotNull(assignedList);


            var details = new List<string> { null, "test" };
            // null and string
            var q2 = db.Task.ByDetails(details);
            var detailList = q2.ToList();
            Assert.IsNotNull(detailList);

            // null and int direct
            var q3 = db.Task.ByAssignedId(null, 1);

            var assignedList2 = q3.ToList();
            Assert.IsNotNull(assignedList2);

            // string, null direct
            var q4 = db.Task.ByDetails("test", null, "other");
            var detailList2 = q4.ToList();
            Assert.IsNotNull(detailList2);

            // should use IN operator
            var q5 = db.Task.ByCreatedId(1, 2);
            var createdList = q5.ToList();
            Assert.IsNotNull(createdList);
        }

        [Test]
        public void ContainmentTest()
        {
            var db = new TrackerDataContext { Log = Console.Out };

            var q1 = db.Task.BySummary(ContainmentOperator.Contains, "Earth");
            var list = q1.ToList();
            Assert.IsNotNull(list);

            var q2 = db.Task.BySummary(ContainmentOperator.StartsWith, "Make");
            list = q2.ToList();
            Assert.IsNotNull(list);

            var q3 = db.Task.BySummary(ContainmentOperator.EndsWith, "Earth");
            list = q3.ToList();
            Assert.IsNotNull(list);

            var q4 = db.Task.BySummary(ContainmentOperator.NotContains, "test");
            list = q4.ToList();
            Assert.IsNotNull(list);

            var q5 = db.Task.ByDetails(ContainmentOperator.Equals, null);
            list = q5.ToList();
            Assert.IsNotNull(list);

            var q6 = db.Task.ByDetails(ContainmentOperator.NotEquals, null);
            list = q6.ToList();
            Assert.IsNotNull(list);
        }

        [Test]
        [ExpectedException(typeof(ArgumentNullException))]
        public void ContainmentStartsWith()
        {
            var db = new TrackerDataContext { Log = Console.Out };

            var q5 = db.Task.ByDetails(ContainmentOperator.StartsWith, null);
            var list = q5.ToList();
            Assert.IsNotNull(list);          
        }

        [Test]
        [ExpectedException(typeof(ArgumentNullException))]
        public void ContainmentEndsWith()
        {
            var db = new TrackerDataContext { Log = Console.Out };

            var q6 = db.Task.ByDetails(ContainmentOperator.EndsWith, null);
            var list = q6.ToList();
            Assert.IsNotNull(list);
        }

        [Test]
        [ExpectedException(typeof(ArgumentNullException))]
        public void ContainmentContains()
        {
            var db = new TrackerDataContext { Log = Console.Out };

            var q8 = db.Task.ByDetails(ContainmentOperator.Contains, null);
            var list = q8.ToList();
            Assert.IsNotNull(list);
        }

        [Test]
        public void ComparisonTest()
        {
            var db = new TrackerDataContext { Log = Console.Out };

            var q1 = db.Task.ByStartDate(ComparisonOperator.GreaterThan, DateTime.Now);
            var list = q1.ToList();
            Assert.IsNotNull(list);

            var q2 = db.Task.ByStartDate(ComparisonOperator.GreaterThanOrEquals, DateTime.Now);
            list = q2.ToList();
            Assert.IsNotNull(list);

            var q3 = db.Task.ByStartDate(ComparisonOperator.LessThan, DateTime.Now);
            list = q3.ToList();
            Assert.IsNotNull(list);

            var q4 = db.Task.ByStartDate(ComparisonOperator.LessThanOrEquals, DateTime.Now);
            list = q4.ToList();
            Assert.IsNotNull(list);

            var q5 = db.Task.ByStartDate(ComparisonOperator.NotEquals, DateTime.Now);
            list = q5.ToList();
            Assert.IsNotNull(list);

            var q6 = db.Task.ByStartDate(ComparisonOperator.Equals, null);
            list = q6.ToList();
            Assert.IsNotNull(list);

            var q7 = db.Task.ByStartDate(ComparisonOperator.NotEquals, null);
            list = q7.ToList();
            Assert.IsNotNull(list);
        }

        [Test]
        [ExpectedException(typeof(ArgumentNullException))]
        public void ComparisonGreaterThan()
        {
            var db = new TrackerDataContext { Log = Console.Out };

            var q6 = db.Task.ByStartDate(ComparisonOperator.GreaterThan, null);
            var list = q6.ToList();
            Assert.IsNotNull(list);
        }

        [Test]
        [ExpectedException(typeof(ArgumentNullException))]
        public void ComparisonGreaterThanOrEquals()
        {
            var db = new TrackerDataContext { Log = Console.Out };

            var q6 = db.Task.ByStartDate(ComparisonOperator.GreaterThanOrEquals, null);
            var list = q6.ToList();
            Assert.IsNotNull(list);
        }

        [Test]
        [ExpectedException(typeof(ArgumentNullException))]
        public void ComparisonLessThan()
        {
            var db = new TrackerDataContext { Log = Console.Out };

            var q6 = db.Task.ByStartDate(ComparisonOperator.LessThan, null);
            var list = q6.ToList();
            Assert.IsNotNull(list);
        }

        [Test]
        [ExpectedException(typeof(ArgumentNullException))]
        public void ComparisonLessThanOrEquals()
        {
            var db = new TrackerDataContext { Log = Console.Out };

            var q6 = db.Task.ByStartDate(ComparisonOperator.LessThanOrEquals, null);
            var list = q6.ToList();
            Assert.IsNotNull(list);
        }
    }
}

File: Framework-Samples/PLINQO/VisualBasic/Tracker/Source/Tracker.Tests/QueryTests/OperatorTests.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports CodeSmith.Data.Linq
Imports NUnit.Framework
Imports Tracker.Core.Data

Namespace Tracker.Tests.QueryTests
    <TestFixture()> _
    Public Class OperatorTests
        Inherits TestBase
        <Test()> _
        Public Sub Contains()
            Dim db As New TrackerDataContext() With { _
             .Log = Console.Out _
            }

            Dim assignedId As New List(Of System.Nullable(Of Integer))()
            assignedId.Add(Nothing)
            assignedId.Add(1)

            ' null and int
            Dim q1 As IQueryable(Of Task) = db.Task.ByAssignedId(assignedId)

            Dim assignedList As List(Of Task) = q1.ToList()
            Assert.IsNotNull(assignedList)


            Dim details As New List(Of String)()
            details.Add(Nothing)
            details.Add("test")

            ' null and string
            Dim q2 As IQueryable(Of Task) = db.Task.ByDetails(details)
            Dim detailList As List(Of Task) = q2.ToList()
            Assert.IsNotNull(detailList)

            ' null and int direct
            Dim q3 As IQueryable(Of Task) = db.Task.ByAssignedId(Nothing, 1)

            Dim assignedList2 As List(Of Task) = q3.ToList()
            Assert.IsNotNull(assignedList2)

            ' string, null direct
            Dim q4 As IQueryable(Of Task) = db.Task.ByDetails("test", Nothing, "other")
            Dim detailList2 As List(Of Task) = q4.ToList()
            Assert.IsNotNull(detailList2)

            ' should use IN operator
            Dim q5 As IQueryable(Of Task) = db.Task.ByCreatedId(1, 2)
            Dim createdList As List(Of Task) = q5.ToList()
            Assert.IsNotNull(createdList)
        End Sub

        <Test()> _
        Public Sub ContainmentTest()
            Dim db As New TrackerDataContext() With { _
             .Log = Console.Out _
            }

            Dim q1 As IQueryable(Of Task) = db.Task.BySummary(ContainmentOperator.Contains, "Earth")
            Dim list As List(Of Task) = q1.ToList()
            Assert.IsNotNull(list)

            Dim q2 As IQueryable(Of Task) = db.Task.BySummary(ContainmentOperator.StartsWith, "Make")
            list = q2.ToList()
            Assert.IsNotNull(list)

            Dim q3 As IQueryable(Of Task) = db.Task.BySummary(ContainmentOperator.EndsWith, "Earth")
            list = q3.ToList()
            Assert.IsNotNull(list)

            Dim q4 As IQueryable(Of Task) = db.Task.BySummary(ContainmentOperator.NotContains, "test")
            list = q4.ToList()
            Assert.IsNotNull(list)

            Dim q5 As IQueryable(Of Task) = db.Task.ByDetails(ContainmentOperator.Equals, Nothing)
            list = q5.ToList()
            Assert.IsNotNull(list)

            Dim q6 As IQueryable(Of Task) = db.Task.ByDetails(ContainmentOperator.NotEquals, Nothing)
            list = q6.ToList()
            Assert.IsNotNull(list)
        End Sub

        <Test()> _
        <ExpectedException(GetType(ArgumentNullException))> _
        Public Sub ContainmentStartsWith()
            Dim db As New TrackerDataContext() With { _
             .Log = Console.Out _
            }

            Dim q5 As IQueryable(Of Task) = db.Task.ByDetails(ContainmentOperator.StartsWith, Nothing)
            Dim list As List(Of Task) = q5.ToList()
            Assert.IsNotNull(list)
        End Sub

        <Test()> _
        <ExpectedException(GetType(ArgumentNullException))> _
        Public Sub ContainmentEndsWith()
            Dim db As New TrackerDataContext() With { _
             .Log = Console.Out _
            }

            Dim q6 As IQueryable(Of Task) = db.Task.ByDetails(ContainmentOperator.EndsWith, Nothing)
            Dim list As List(Of Task) = q6.ToList()
            Assert.IsNotNull(list)
        End Sub

        <Test()> _
        <ExpectedException(GetType(ArgumentNullException))> _
        Public Sub ContainmentContains()
            Dim db As New TrackerDataContext() With { _
             .Log = Console.Out _
            }

            Dim q8 As IQueryable(Of Task) = db.Task.ByDetails(ContainmentOperator.Contains, Nothing)
            Dim list As List(Of Task) = q8.ToList()
            Assert.IsNotNull(list)
        End Sub

        <Test()> _
        Public Sub ComparisonTest()
            Dim db As New TrackerDataContext() With { _
             .Log = Console.Out _
            }

            Dim q1 As IQueryable(Of Task) = db.Task.ByStartDate(ComparisonOperator.GreaterThan, DateTime.Now)
            Dim list As List(Of Task) = q1.ToList()
            Assert.IsNotNull(list)

            Dim q2 As IQueryable(Of Task) = db.Task.ByStartDate(ComparisonOperator.GreaterThanOrEquals, DateTime.Now)
            list = q2.ToList()
            Assert.IsNotNull(list)

            Dim q3 As IQueryable(Of Task) = db.Task.ByStartDate(ComparisonOperator.LessThan, DateTime.Now)
            list = q3.ToList()
            Assert.IsNotNull(list)

            Dim q4 As IQueryable(Of Task) = db.Task.ByStartDate(ComparisonOperator.LessThanOrEquals, DateTime.Now)
            list = q4.ToList()
            Assert.IsNotNull(list)

            Dim q5 As IQueryable(Of Task) = db.Task.ByStartDate(ComparisonOperator.NotEquals, DateTime.Now)
            list = q5.ToList()
            Assert.IsNotNull(list)

            Dim q6 As IQueryable(Of Task) = db.Task.ByStartDate(ComparisonOperator.Equals, Nothing)
            list = q6.ToList()
            Assert.IsNotNull(list)

            Dim q7 As IQueryable(Of Task) = db.Task.ByStartDate(ComparisonOperator.NotEquals, Nothing)
            list = q7.ToList()
            Assert.IsNotNull(list)
        End Sub

        <Test()> _
        <ExpectedException(GetType(ArgumentNullException))> _
        Public Sub ComparisonGreaterThan()
            Dim db As New TrackerDataContext() With { _
             .Log = Console.Out _
            }

            Dim q6 As IQueryable(Of Task) = db.Task.ByStartDate(ComparisonOperator.GreaterThan, Nothing)
            Dim list As List(Of Task) = q6.ToList()
            Assert.IsNotNull(list)
        End Sub

        <Test()> _
        <ExpectedException(GetType(ArgumentNullException))> _
        Public Sub ComparisonGreaterThanOrEquals()
            Dim db As New TrackerDataContext() With { _
             .Log = Console.Out _
            }

            Dim q6 As IQueryable(Of Task) = db.Task.ByStartDate(ComparisonOperator.GreaterThanOrEquals, Nothing)
            Dim list As List(Of Task) = q6.ToList()
            Assert.IsNotNull(list)
        End Sub

        <Test()> _
        <ExpectedException(GetType(ArgumentNullException))> _
        Public Sub ComparisonLessThan()
            Dim db As New TrackerDataContext() With { _
             .Log = Console.Out _
            }

            Dim q6 As IQueryable(Of Task) = db.Task.ByStartDate(ComparisonOperator.LessThan, Nothing)
            Dim list As List(Of Task) = q6.ToList()
            Assert.IsNotNull(list)
        End Sub

        <Test()> _
        <ExpectedException(GetType(ArgumentNullException))> _
        Public Sub ComparisonLessThanOrEquals()
            Dim db As New TrackerDataContext() With { _
             .Log = Console.Out _
            }

            Dim q6 As IQueryable(Of Task) = db.Task.ByStartDate(ComparisonOperator.LessThanOrEquals, Nothing)
            Dim list As List(Of Task) = q6.ToList()
            Assert.IsNotNull(list)
        End Sub
    End Class
End Namespace
